### PR TITLE
Breaks Captcha

### PR DESCRIPTION
**Pull request by MeaningOf42**
Thanks to @jmrushing for providing the code to break the captcha. Just tried to make the code a bit neater. It now has a separate function for breaking the captcha and calls on FFmpeg without a wrapper library as this appeared easier than getting everyone to sort out the extra dependencies. Additionally, it tells people to install FFmpeg in the read me. (Also thanks to @joeyagreco I'd started working on my PR before you submitted yours but I used yours to speed up adding the new constants to my code.)

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -1,26 +1,113 @@
...
+            return(text)     
+        except Exception as e:
+            print(e)
+            print('Sorry.. run again...')
+
+def saveFile(content,filename):
+    with open(filename, "wb") as handle:
+        for data in content.iter_content():
+            handle.write(data)
+# END TEST
```
Comment (joeyagreco) [inline main.py:57]: can you remove this

File: main.py [hunk 2]
```diff
@@ -1,26 +1,113 @@
...
+
+def solveCaptcha(driver):
+    # Logic to click through the reCaptcha to the Audio Challenge, download the challenge mp3 file, run it through the audioToText function, and send answer
+    googleClass = driver.find_elements_by_class_name(CAPTCHA_BOX)[0]
+    time.sleep(2)
+    outeriframe = googleClass.find_element_by_tag_name('iframe')
+    time.sleep(1)
+    outeriframe.click()
+    time.sleep(2)
+    allIframesLen = driver.find_elements_by_tag_name('iframe')
```
Comment (joeyagreco) [inline main.py:67]: "allIFrames" seems like a more accurate name for this variable, unless you want to just save the len() of this directly and skip it on line 71

**Review by joeyagreco — APPROVED** [review]
Suggested a few minor changes, otherwise looks good!

File: main.py [hunk 3]
```diff
@@ -1,26 +1,113 @@
...
+r = sr.Recognizer()
+
+def audioToText(mp3Path):
+    # deletes old file
+    try:
+        os.remove(CAPTCHA_WAV_FILENAME)
+    except FileNotFoundError:
+        pass
+    # convert wav to mp3                                                            
+    subprocess.run(f"ffmpeg -i {mp3Path} {CAPTCHA_WAV_FILENAME}", shell=True, timeout=5)
```
Comment (jmrushing) [inline main.py:41]: Are there any benefits to using ffmpeg over pydub's  AudioSegment? Seems like an extra layer of complexity for the end user (installing ffmpeg & setting up the path) vs 3 lines of code in main.py and 1 in requirements.txt à la #26 .
A better way than both methods would be probably be to save the incoming stream straight to WAV, but my brain is too fried to work that out at the moment.
Also, is checking for the file and deleting necessary before running ffmpeg? Using the solution in #26, `sound.export()` overwrites the old WAV file during each new iteration, perhaps ffmpeg can't do that for some reason (permissions?) or needs an additional command flag?

Comment (MeaningOf42) [reply main.py:41]: I simply couldn't get pydubs working. I think you are right though saving directly to wav would be best

Comment (MeaningOf42) [reply main.py:41]: Also my understanding is that pydubs relies on ffmpeg for conversion anyway

File: main.py (hunk 2 above)
Comment (MeaningOf42) [reply main.py:67]: Yeah you are correct there

File: main.py (hunk 3 above)
Comment (jmrushing) [reply main.py:41]: I see. I wonder if anyone else who's tested with the #26 solution has had issues with pydub? I hope not... :facepalm: 
I didn't mean to give you hard time about ffmpeg, but I've done A LOT of scripting with it in the past. And while it's been...a while...for me, ffmpeg used to have a pretty dodgy reputation for breaking things with their updates, in addition to some nasty security loopholes stemming from outdated libraries. Maybe that's not the case anymore, but it was bad enough just a few years ago that I would regularly have to re-write scripts after updates to account for altered internal default values or implementation differences between distros. That's my unsolicited $0.02 on ffmpeg :sweat_smile:

Comment (jmrushing) [reply main.py:41]: > Also my understanding is that pydubs relies on ffmpeg for conversion anyway

Well I learned [something](https://github.com/jiaaro/pydub#dependencies) today. I should have RTFM. Doh!

ffmpeg on its own is still kludgy IMO, maybe that's why I prefer using the pydub wrapper. When I get some time, I'll work out the "clean" solution and hopefully we'll be done with creating multiple audio files.

**Pull request merged**